ddtrace/mocktracer/civisibilitymocktracer.go

```diff
@@ -0,0 +1,178 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package mocktracer
+
+import (
+	"sync"
+	"sync/atomic"
+	_ "unsafe" // for go:linkname
+
+	"github.com/DataDog/dd-trace-go/v2/ddtrace/ext"
+	"github.com/DataDog/dd-trace-go/v2/ddtrace/tracer"
+	"github.com/DataDog/dd-trace-go/v2/internal/civisibility"
+	"github.com/DataDog/dd-trace-go/v2/internal/civisibility/constants"
+	"github.com/DataDog/dd-trace-go/v2/internal/datastreams"
+)
+
+//go:linkname storeGlobalTracer github.com/DataDog/dd-trace-go/v2/ddtrace/tracer.storeGlobalTracer
+func storeGlobalTracer(t tracer.Tracer)
```

Comment:
This will be turned obsolete once #3482 is approved and merged.

Reply:
ohh, you changed everything 😄 , I'll wait for your PR to be merged first.